examples/xlnet/xlnet_generation_main.py

```diff
@@ -0,0 +1,145 @@
+                    default="pretrained/xlnet_cased_L-24_H-1024_A-16/"
+                            "spiece.model",
+                    help="Location to load sentene piece model from")
+
+args = parser.parse_args()
+
+
+def main():
+    pretrain_checkpoint = args.pretrain_checkpoint
+    sentence_piece_model = args.sentence_piece
+    if torch.cuda.is_available():
+        device = torch.device(torch.cuda.current_device())
+    else:
+        device = 'cpu'
+
+    model = xlnet.model.decoder.XLNetDecoder()
+    xlnet.model.load_from_tf_checkpoint(model, pretrain_checkpoint)
+    model = model.to(device)
+    sp_model = spm.SentencePieceProcessor()
+    sp_model.Load(sentence_piece_model)
+    tokenize_fn = xlnet.data.create_tokenize_fn(sp_model, False)
+
+    # A lengthy padding text used to workaround lack of context for short
+    # prompts. Refer to https://github.com/rusiaaman/XLNet-gen for the rationale
+    # behind this.
+    pad_txt = """
+        In 1991, the remains of Russian Tsar Nicholas II and his family
+        (except for Alexei and Maria) are discovered.
+        The voice of Nicholas's young son, Tsarevich Alexei Nikolaevich,
+        narrates the remainder of the story. 1883 Western Siberia, a young
+        Grigori Rasputin is asked by his father and a group of men to perform
+        magic. Rasputin has a vision and denounces one of the men as a horse
+        thief. Although his father initially slaps him for making such an
+        accusation, Rasputin watches as the man is chased outside and beaten.
+        Twenty years later, Rasputin sees a vision of the Virgin Mary,
+        prompting him to become a priest. Rasputin quickly becomes famous,
+        with people, even a bishop, begging for his blessing. """
+    pad_ids = tokenize_fn(pad_txt)
+    pad_ids.append(xlnet.data.utils.EOD_ID)
+
+    def split_by(xs, y):
+        p = 0
+        for idx, x in enumerate(xs):
+            if x == y:
+                if idx - p > 0:
+                    yield xs[p:idx]
+                p = idx + 1
+        if len(xs) - p > 0:
+            yield xs[p:]
+
+    def sample(text: str, length: int = 200, n_samples=3, **kwargs):
+        print("=== Prompt ===")
+        print(text)
+        model.eval()
+        text = text.replace("\n", "<eop>")
+        tokens = pad_ids + tokenize_fn(text)
+        tokens = torch.tensor(tokens, device=device).expand(n_samples, -1)
+        kwargs.setdefault("print_steps", True)
+        decode_output, _ = model(
+            tokens, max_decoding_length=length, **kwargs)
+        decode_samples = decode_output.sample_id.tolist()
+        for idx, sample_tokens in enumerate(decode_samples):
+            print(f"=== Sample {idx} ===")
+            output = "\n".join(sp_model.DecodeIds(xs) for xs in split_by(
+                sample_tokens, xlnet.data.utils.special_symbols["<eop>"]))
+            print(output)
+
+    if args.is_interactive:
+        try:
+            from IPython import embed
```

Comment:
There shouldn't be `IPython`. Please make everything the same as [gpt2_generate_main.py](https://github.com/asyml/texar-pytorch/blob/8ce8589120dc54bcd28c8f9458128422f998bdf2/examples/gpt-2/gpt2_generate_main.py), except that here we use XLNet instead of GPT-2.

Reply:
E.g., In gpt2-generate-main, https://github.com/asyml/texar-pytorch/blob/8ce8589120dc54bcd28c8f9458128422f998bdf2/examples/gpt-2/gpt2_generate_main.py#L109-L112

Users only type in the prompt text (instead of the code `sample(...)`)

Reply:
GPT2 has interactive option. Should this file also have interactive option but without using IPython?

Reply:
Got it.

Reply:
yep. we could just copy-paste the interactive code from gpt2

Reply:
Removed IPython dependency in f3ceca0